ballerina/types.bal

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2025 WSO2 LLC. (http://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+
+# Represents any error related to Ballerina Milvus connector.
+public type Error distinct error;
+
+# Represents the configuration for the Milvus connection.
+#
+# + username - The username for the Milvus connection
+# + password - The password for the Milvus connection
+# + databaseName - The database name for the Milvus connection
+public type ConnectionConfig record {
+    *http:ClientConfiguration;
+    string username?;
+    string password?;
+    string databaseName?;
+};
+
+# Represents the request for the upsert operation.
+#
+# + collectionName - The name of the collection to upsert data into
+# + partitionName - The name of the partition to upsert data into
+# + databaseName - The name of the database to upsert data into
+# + primaryKey - The name of the primary key of the collection (default is `id`)
+# + data - The data to upsert into the Milvus collection
+public type UpsertRequest record {
+    string collectionName;
+    string partitionName?;
+    string databaseName?;
+    record {
+        string id;
+        float[] vectors;
+    } data;
+};
+
+# Represents the request for the delete operation.
+#
+# + collectionName - The name of the collection to delete data from
+# + partitionName - The name of the partition to delete data from
+# + id - The id of the data to delete
+# + filter - The filter to delete data from the Milvus collection
+public type DeleteRequest record {
+    string collectionName;
+    string partitionName?;
+    int[] id?;
```

Comment:
```suggestion
    int[] ids?;
```

Reply:
Fixed in [abfa402](https://github.com/ballerina-platform/module-ballerinax-milvus/pull/1/commits/abfa4023b796ed5ccf9a09e858e8a68feb6b9b90)